rcore/appmanager.c

```diff
@@ -115,39 +116,80 @@ static App *_appmanager_create_app(char *name, uint8_t type, void *entry_point,
     app->type = type;
     app->header = NULL;
     app->next = NULL;
-    app->slot_id = slot_id;
+    app->app_file = app_file;
+    app->resource_file = resource_file;
     app->is_internal = is_internal;
     
     return app;
 }
 
 
+struct appdb
+{
+    uint32_t paged_incr;  //0x58F6AExx
+    uint16_t app_cache_id; //  B5 3E trek v2 also appears in app cache
+    uint16_t unk;  // 0x0FC13Exx
+    uint32_t application_id; // in app as @000037f
+    Uuid app_uuid;  // 16 bytes
+    uint32_t unk_13;
+    uint32_t unk_17;
+    uint16_t sdk_version;
+    uint16_t app_version;
+    uint16_t zeros;
+    uint8_t app_name[32];
+    uint8_t unk_arr_company[32];  // always blank
+    uint8_t unk_arr[32]; // always blank
+} __attribute__((__packed__));
+
+
 /*
  * Load the list of apps and faces from flash
  * The app manifest is a list of all known applications we found in flash
- * We scan all block regions and look for app signatures.
- * TODO The real app likely does nothing quite so crude. We need to find the app table!
+ * We load all entries from `appdb` file.
+ * TODO: appdb seems to have duplicates (in my case), maybe we should use `pmap`, but it was missing some entries for me
  */
 void _appmanager_flash_load_app_manifest(void)
 {
+    File appdb_file;
```

Comment:
Hmm, can we change `file` to be `struct file`, while we're here?  `File` to me reads like 'small primitive type', and the `struct` keyword before tells me that it is a larger type.

Reply:
That's exactly why I like uppercase names, big letter = big type :), and then don't need to write ugly `struct` everywhere 
but sure, that's just a style preference from objective languages